### PR TITLE
Transform benchmark resnext50 changing to models

### PR DESCRIPTION
**Pull request by zhaoyuchen2018**
This changing can improve performance

test=develop

### CODE REVIEW

File: PaddleCV/image_classification/train.py [hunk 1]
```diff
@@ -7,6 +7,22 @@
...
+        if os.environ.get(key, None) is None:
+            os.environ[key] = str(value)
+
+
+# NOTE(paddle-dev): All of these flags should be
+# set before `import paddle`. Otherwise, it would
+# not take any effect. 
+set_paddle_flags({
+    'FLAGS_eager_delete_tensor_gb': 0,  # enable gc 
+    'FLAGS_memory_fraction_of_eager_deletion': 0.98
```
Comment (sneaxiy) [inline PaddleCV/image_classification/train.py]: `FLAGS_fraction_of_gpu_memory_to_use`? See [here](https://github.com/PaddlePaddle/benchmark/blob/master/se-resnext/paddle/run.sh#L8).

Comment (zhaoyuchen2018) [reply PaddleCV/image_classification/train.py]: thanks，fixed

**Review by heavengate — APPROVED** [review]
LGTM

File: PaddleCV/image_classification/train.py [hunk 2]
```diff
@@ -412,10 +426,20 @@ def if_exist(var):
     # use_ngraph is for CPU only, please refer to README_ngraph.md for details
     use_ngraph = os.getenv('FLAGS_use_ngraph')
     if not use_ngraph:
+        build_strategy = fluid.BuildStrategy()
+        build_strategy.memory_optimize = args.with_mem_opt
+        build_strategy.enable_inplace = args.with_inplace
+        build_strategy.fuse_all_reduce_ops=1
```
Comment (qingqing01) [inline PaddleCV/image_classification/train.py:432]: 这个参数测过了吗？ 似乎QA测试出错了

File: PaddleCV/image_classification/train.py [hunk 3]
```diff
@@ -412,10 +426,20 @@ def if_exist(var):
     # use_ngraph is for CPU only, please refer to README_ngraph.md for details
     use_ngraph = os.getenv('FLAGS_use_ngraph')
     if not use_ngraph:
+        build_strategy = fluid.BuildStrategy()
+        build_strategy.memory_optimize = args.with_mem_opt
```
Comment (qingqing01) [inline PaddleCV/image_classification/train.py:430]: 现在是默认mem_opt和GC同时开？这样有没有风险？

File: PaddleCV/image_classification/train.py [hunk 4]
```diff
@@ -50,6 +63,7 @@
 add_arg('image_shape',      str,   "3,224,224",          "input image size")
 add_arg('model_save_dir',   str,   "output",             "model save directory")
 add_arg('with_mem_opt',     bool,  True,                 "Whether to use memory optimization or not.")
+add_arg('with_inplace',     bool,  True,                 "Whether to use inplace memory optimization.")
```
Comment (shippingwang) [inline PaddleCV/image_classification/train.py:66]: Plz add args explaination in README when adding a new args

**Pull request merged**